docs/envoy.md

```diff
@@ -0,0 +1,326 @@
+# Life of an OSM Request
+
+## A bit on Envoy
+
+OSM heavily uses envoy for proxying requests, applying mTLS, and exporting statistics. Envoy itself has 5 main
+configuration mechanisms for request manipulation:
+
+* ListenerConfigs
+* RouteConfigs
+* EndpointConfigs
+* ClusterConfigs
+* SecretConfigs
+
+### Listeners
+
+Envoy itself intercepts all requests by reconfiguring the IP tables on the Pod to reroute incoming and outbound traffic
+to the Sidecar. In OSM, incoming requests are intercepted and forwarded to port 15003, while outbound requests are
+forwarded to port 15001. Each port is configured with its own listener.
+
+In addition to the address and port to listen on, a Listener is configured with a set of `filter_chains` which dictate
+what to look for and what to do with a matching request, and a `listener_filter`, which tells Envoy what data it needs
+to extract from the request.
+
+Each Listener has a:
+
+
+Each listener will have a single `filter_chain` for every service. That is, the inbound listener will have 1 filter
+chain for each service that can reference its associated Pod, and the outbound filter will have 1 filter chain for every
```

Comment:
This part is not correct. Both inbound and outbound filter chains are per service per port.

Reply:
You're right, see that in the code now.. will fix shortly, however does that mean that there is currently a bug in naming the filterchain?
Since the name is supposed to be unique
https://github.com/envoyproxy/envoy/blob/123e81813a2003de07c8e3400c090bb755676a63/api/envoy/config/listener/v3/listener_components.proto#L256, yet it's scoped to the service https://github.com/openservicemesh/osm/blob/main/pkg/envoy/lds/inmesh.go#L312.

 not sure if this results in a bug or not but wanted to check here.